depthai_ros_driver/src/conversion.cpp

```diff
@@ -0,0 +1,75 @@
+#include "depthai_ros_driver/conversion.hpp"
+#include <opencv2/imgproc.hpp>
+
+cv::Mat rr::convertImg(const depthai_datatype_msgs::RawImgFrame& input) {
+    // Note: stolen from depthai-core/src/opencv/ImgFrame.cpp
+    cv::Mat mat;
+    cv::Size size = {0, 0};
+    int type = 0;
+
+    switch (static_cast<dai::RawImgFrame::Type>(input.fb.type)) {
+        case dai::RawImgFrame::Type::RGB888i:
+        case dai::RawImgFrame::Type::BGR888i:
+        case dai::RawImgFrame::Type::BGR888p:
+        case dai::RawImgFrame::Type::RGB888p:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_8UC3;
+            break;
+
+        case dai::RawImgFrame::Type::YUV420p:
+        case dai::RawImgFrame::Type::NV12:
+        case dai::RawImgFrame::Type::NV21:
+            size = cv::Size(input.fb.width, input.fb.height * 3 / 2);
+            type = CV_8UC1;
+            break;
+
+        case dai::RawImgFrame::Type::RAW8:
+        case dai::RawImgFrame::Type::GRAY8:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_8UC1;
+            break;
+
+        case dai::RawImgFrame::Type::GRAYF16:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_16FC1;
+            break;
+
+        case dai::RawImgFrame::Type::RAW16:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_16UC1;
+            break;
+
+        case dai::RawImgFrame::Type::RGBF16F16F16i:
+        case dai::RawImgFrame::Type::BGRF16F16F16i:
+        case dai::RawImgFrame::Type::RGBF16F16F16p:
+        case dai::RawImgFrame::Type::BGRF16F16F16p:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_16FC3;
+            break;
+
+        case dai::RawImgFrame::Type::BITSTREAM:
+        default:
+            size = cv::Size(static_cast<int>(input.data.size()), 1);
+            type = CV_8UC1;
+            break;
+    }
+
+    // Check if enough data
+    long requiredSize = CV_ELEM_SIZE(type) * size.area();
+    long actualSize = static_cast<long>(input.data.size());
+    if (actualSize < requiredSize) {
+        ROS_WARN_STREAM("ImgFrame doesn't have enough data to encode specified frame, required "
+                        << requiredSize << ", actual " << actualSize << ". Maybe metadataOnly transfer was made?");
+    }
+    if (input.fb.width <= 0 || input.fb.height <= 0) {
+        ROS_WARN("ImgFrame metadata not valid (width or height = 0)");
+    }
+
+    mat.create(size, type);
+    std::memcpy(
```

Comment:
asserting `(input.data.size()) == (mat.dataend - mat.datastart)` and making the conversion to fail will prevent wrong copy, and make us notice the error. 
Data size of cv::Mat can be larger than width x height if image width is an odd number, but I don't think this case will not happen in using depth AI image.

Reply:
added assert and simplify the amount of memcpy